impl/src/main/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGenerator.java

```diff
@@ -27,24 +26,42 @@
 public class KeytoolCertificateGenerator {
     private String password;
     private final Path certFilePath;
+    private final Path keyStoreFilePath;
+    private final Path trustStoreFilePath;
     private final System.Logger log = System.getLogger(KeytoolCertificateGenerator.class.getName());
 
-    public KeytoolCertificateGenerator() {
-        Path certsDirectory = null;
-        try {
-            certsDirectory = Files.createTempDirectory("kproxy");
+    public KeytoolCertificateGenerator() throws IOException {
+        this(null, null);
+    }
+
+    public KeytoolCertificateGenerator(String certFilePath, String trustStorePath) throws IOException {
+        Path certsDirectory = Files.createTempDirectory("kproxy");
+        this.certFilePath = Path.of(certsDirectory.toAbsolutePath() + "/cert-file");
+        this.keyStoreFilePath = (certFilePath != null) ? Path.of(certFilePath) : Paths.get(certsDirectory.toAbsolutePath().toString(), "kafka.keystore.jks");
+        this.trustStoreFilePath = (trustStorePath != null) ? Path.of(trustStorePath) : Paths.get(certsDirectory.toAbsolutePath().toString(), "kafka.truststore.jks");
+
+        certsDirectory.toFile().deleteOnExit();
+        if(certFilePath == null) {
+            this.keyStoreFilePath.toFile().deleteOnExit();
         }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
+        if(trustStorePath == null) {
+            this.trustStoreFilePath.toFile().deleteOnExit();
```

Comment:
nit: I'm confused why we test if the `trustStorePath` is null and then mark `this.trustStoreFilePath` as delete on exit.

Reply:
Because in the constructor the kafka client can set its own trustStorePath and we won't delete this file. If `null` we are sure that the trustStoreFilePath is temporary and we can delete it.

Reply:
It's probably worth a comment to that effect, given the odds on me remembering that are slim...